Kudu.FunctionalTests/OneDeployTests.cs

```diff
@@ -0,0 +1,389 @@
+        [InlineData("false")]
+        public Task TestZipDeployment(string async)
+        {
+            return ApplicationManager.RunAsync("TestZipDeployment", async appManager =>
+            {
+                var files = CreateRandomFilesForZip(10);
+                var response = await DeployZippedArtifact(appManager, files, new ZipDeployMetadata(), "zip", null, async);
+                response.EnsureSuccessStatusCode();
+
+                if (async == "true")
+                {
+                    DeployResult result;
+                    do
+                    {
+                        result = await appManager.DeploymentManager.GetResultAsync("latest");
+                        Assert.Equal(deployer, result.Deployer);
+                        await Task.Delay(TimeSpan.FromSeconds(2));
+                    } while (!new[] { DeployStatus.Failed, DeployStatus.Success }.Contains(result.Status));
+                }
+
+                await AssertSuccessfulDeploymentByFilenames(appManager, files.Select(f => f.Filename).ToArray());
+            });
+        }
+
+        [Theory]
+        [InlineData("true")]
+        [InlineData("false")]
+        public Task TestURLBasedDeployment(string async)
+        {
+            var packageUri = "https://github.com/dannysongg/testrepo/blob/master/green.war?raw=true";
+            return ApplicationManager.RunAsync("TestURLDeployment", async appManager =>
+            {
+                var client = appManager.OneDeployManager.Client;
+                var requestUri = client.BaseAddress + "?type=war&async=" + async;
+                using (var request = new HttpRequestMessage(HttpMethod.Put, requestUri))
+                {
+                    var payload = new { packageUri = packageUri };
+                    request.Content = new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json");
+
+                    using (var response = await client.SendAsync(request))
+                    {
+                        if (async == "true")
+                        {
+                            DeployResult result;
+                            do
+                            {
+                                result = await appManager.DeploymentManager.GetResultAsync("latest");
+                                Assert.Equal(deployer, result.Deployer);
+                                await Task.Delay(TimeSpan.FromSeconds(2));
+                            } while (!new[] { DeployStatus.Failed, DeployStatus.Success }.Contains(result.Status));
+                        }
+                        else
+                        {
+                            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        }
+                        await AssertSuccessfulDeploymentByFilenames(appManager, new string[] { "app.war" });
+                    }
+                }
+
+            });
+        }
+
+        [Fact]
+        public Task EnsureIterativeDeployment()
```

Comment:
"Incremental"Deployment instead of "Iterative"Deployment. Also, use 'Test' as a prefix for all Tests, instead of 'Ensure'